[libc++] [string_view] Remove operators made redundant by C++20

AMP999 opened the pull request:
Thanks to Giuseppe D'Angelo for pointing this out on the cpplang Slack!

The example implementation here...
https://eel.is/c++draft/string.view.comparison#example-1 ...was necessary when it was written, in C++17, but in C++20 we don't need that complexity anymore, because of the reversed candidates that are synthesized by the compiler.

ldionne commented on libcxx/include/string_view:
```diff
@@ -776,7 +776,33 @@ template <ranges::contiguous_range _Range>
 
 // [string.view.comparison]
-// operator ==
+
```
Did you confirm that we had tests that checked for all the ways string views can be compared (i.e. https://eel.is/c++draft/string.view.comparison#tab:string.view.comparison.overloads)?

AMP999 replied:
Yes, see libcxx/test/std/strings/string.view/string.view.comparison/*.pass.cpp

ldionne replied:
Cool, thanks. I'll leave it to @Zingam to say if this looks fine to him but this LGTM.

ldionne reviewed (commented):
This seems reasonable but I'd like @Zingam to take a look since he's done a lot of our support for `operator<=>`.

ldionne commented on libcxx/include/string_view:
```diff
@@ -776,7 +776,35 @@ template <ranges::contiguous_range _Range>
 // [string.view.comparison]
+
+#if _LIBCPP_STD_VER >= 20
+
```
Let's regroup all the C++ >= 20 definitions into the same block. There's a bunch of those below. There's also conditions that can be simplified now, e.g. some `#if _LIBCPP_STD_VER < 20` that would be made tautological since they'd be in the `#else`.

AMP999 replied:
I pushed a wrong commit in the branch. It's fixed now.

AMP999 committed:
[libc++] [string_view] Remove operators made redundant by C++20

Thanks to Giuseppe D'Angelo for pointing this out on the cpplang Slack!

The example implementation here...
https://eel.is/c++draft/string.view.comparison#example-1
...was necessary when it was written, in C++17, but in C++20
we don't need that complexity anymore, because of the reversed
candidates that are synthesized by the compiler.

ldionne commented:
Thanks! FYI @AMP999 I see that your github setting are marked as hiding your email, which means that patches are being committed with a github-generated email address. Just a heads up in case that's not the behavior you want.

AMP999 commented:
You're welcome! Thank you for notifying me about the email!
